### PR TITLE
First pass at openshift multitenancy support for monitoring

### PR DESCRIPTION
**Pull request by BryanEllington**

### CODE REVIEW

**Comment by BryanEllington** [comment]
Feature-flagged this content. Official support is planned for a future release.

File: monitoring/bin/remove_monitoring_tenant.sh [hunk 1]
```diff
@@ -66,4 +74,9 @@ kubectl delete -n $VIYA_NS --ignore-not-found secret prometheus-federate-$VIYA_T
 # Remove non-user-provided certificates
 kubectl delete -n $VIYA_NS --ignore-not-found certificate -l "v4m.sas.com/tenant=$VIYA_TENANT" 2>/dev/null
 
+if [ "$OPENSHIFT_CLUSTER" == "true" ]; then
```
Comment (gsmith-sas) [inline monitoring/bin/remove_monitoring_tenant.sh:77]: Is this the only "OpenShift" specific task needed for removing tenant monitoring?  If so, why have the separate remove_monitoring_tenant_openshift.sh script?  If the other script does "more", why do anything "OpenShift" specific here?

File: monitoring/bin/remove_monitoring_tenant.sh [hunk 2]
```diff
@@ -12,6 +12,14 @@ cd "$(dirname $BASH_SOURCE)/../.."
 source monitoring/bin/common.sh
 source bin/openshift-include.sh
 
+if [ "$OPENSHIFT_CLUSTER" == "true" ]; then  
+  if [ "${CHECK_OPENSHIFT_CLUSTER:-true}" == "true" ]; then
+    log_error "This script should not be run on OpenShift clusters"
```
Comment (gsmith-sas) [inline monitoring/bin/remove_monitoring_tenant.sh:17]: Probably should be reworded to make it more clear that it appears (or we believe) that the current environment is OpenShift.

File: monitoring/bin/deploy_monitoring_tenant_openshift.sh [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+source monitoring/bin/common.sh
+
+if [ "$OPENSHIFT_TENANT_MONITORING_ENABLE" != "true" ]; then
+  log_error "OpenShift tenant monitoring is currently under development and is not supported"
+  exit 1
+fi
+
+if [ "$OPENSHIFT_CLUSTER" != "true" ]; then
+  if [ "${CHECK_OPENSHIFT_CLUSTER:-true}" == "true" ]; then
+    log_error "This script should only be run on OpenShift clusters"
```
Comment (gsmith-sas) [inline monitoring/bin/deploy_monitoring_tenant_openshift.sh:16]: Reword to make it clear that it appears that the script is being run on a non-OpenShift cluster.

File: monitoring/bin/deploy_monitoring_tenant_openshift.sh [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  --atomic \
+  $grafanaPwd \
+  $extraArgs \
+  v4m-grafana-$VIYA_NS-$VIYA_TENANT \
+  grafana/grafana
+
+if [ "$OPENSHIFT_AUTH_ENABLE" == "true" ]; then
+  log_info "Using OpenShift authentication for [$VIYA_TENANT] tenant instance of Grafana"
+  log_debug "Annotating grafana-serviceaccount-$VIYA_TENANT"
+  kubectl annotate serviceaccount -n $VIYA_NS --overwrite grafana-serviceaccount-$VIYA_TENANT 'serviceaccounts.openshift.io/oauth-redirectreference.primary={"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"v4m-grafana-'$VIYA_TENANT'"}}'
```
Comment (gsmith-sas) [inline monitoring/bin/deploy_monitoring_tenant_openshift.sh:168]: Will the OpenShift authentication ensure that tenant admins can ONLY access the tenant-specific instance of these monitoring components and nothing else?  How are tenant admins limited within the OpenShift auth configuration?

**Pull request merged**